addons/sale_management/models/sale_order_template.py

```diff
@@ -29,6 +29,14 @@ def _get_default_require_payment(self):
         domain=[('model', '=', 'sale.order')],
         help="This e-mail template will be sent on confirmation. Leave empty to send nothing.")
     active = fields.Boolean(default=True, help="If unchecked, it will allow you to hide the quotation template without removing it.")
+    company_id = fields.Many2one('res.company', string='Company')
+
+    @api.constrains('company_id', 'sale_order_template_line_ids')
+    def _check_company_id(self):
+        for template in self:
+            for line in template.sale_order_template_line_ids:
+                if line.product_id.company_id and line.product_id.company_id != template.company_id:
+                    raise ValidationError(_("Your quotation template belongs to company %s and cannot contains products restricted to another company (%s belonging to %s). You should use another product or share the product with all companies." % (template.company_id.name, line.product_id.display_name, line.product_id.company_id.name)))
```

Comment:
Can we make this more readable ? (Instead of long text)

![screenshot-551261-34342-0b6909 runbot22 odoo com-2019 06 28-23-06-16](https://user-images.githubusercontent.com/16624719/60360770-2699d400-99fa-11e9-82d0-37428a599f3c.jpg)